src/nvim/undo.c

```diff
@@ -839,7 +839,7 @@ static u_header_T *unserialize_uhp(bufinfo_T *bi, char_u *file_name)
   for (;; ) {
     int len = undo_read_byte(bi);
 
-    if (len == 0) {
+    if (len <= 0) {
       break;
     }
```

Comment:
If we are going to consider this a "corruption", we should behave like line 824:

```
   if (len == EOF) {
     corruption_error("undo_read_byte EOF", file_name);
     xfree(uhp);
     return NULL;
   }
```